src/Control/Effect/Choose.hs

```diff
@@ -1,4 +1,14 @@
 {-# LANGUAGE DeriveGeneric, DeriveTraversable, FlexibleContexts, FlexibleInstances, MultiParamTypeClasses, RankNTypes, TypeOperators, UndecidableInstances #-}
+-- | An effect modelling nondeterminism without failure (one or more successful results).
+--
+-- The 'Control.Effect.NonDet.NonDet' effect is the composition of 'Choose' and
+-- 'Control.Effect.Empty.Empty'.
+--
+-- Predefined carriers:
+--
+-- * "Control.Carrier.Choose.Church.ChooseC".
+-- * If 'Choose' is the last effect in a stack, it can be interpreted directly to a 'NonEmpty'.
```

Comment:
This is pending #250; if there is an objection to #250, I will remove this line.